extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/textTrackPanel.js

```diff
@@ -0,0 +1,170 @@
+/**
+ * Display component for video text tracks
+ *
+ * @module TextTrackPanel
+ */
+var React = require('react');
+var baseFontSize = 1.0;
+
+var TextTrackPanel = React.createClass({
+
+  colorMap: {
+    "White": "255,255,255",
+    "Blue": "0,0,255",
+    "Magenta": "255,0,255",
+    "Green": "0,255,0",
+    "Yellow": "255,255,0",
+    "Red": "255,0,0",
+    "Cyan": "0,255,255",
+    "Black": "0,0,0",
+    "Transparent": "0,0,0"
+  },
+
+  fontTypeMap: {
+    "Monospaced Serif": '"Courier New", Courier, "Nimbus Mono L", "Cutive Mono", monospace',
+    "Proportional Serif": '"Times New Roman", Times, Georgia, Cambria, "PT Serif Caption", serif',
+    "Monospaced Sans-Serif": '"Deja Vu Sans Mono", "Lucida Console", Monaco, Consolas, "PT Mono", monospace',
+    "Proportional Sans-Serif": 'Roboto, "Arial Unicode Ms", Arial, Helvetica, Verdana, "PT Sans Caption", sans-serif',
+    "Casual": '"Comic Sans MS", Impact, Handlee, fantasy',
+    "Cursive": '"Monotype Corsiva", "URW Chancery L", "Apple Chancery", "Dancing Script", cursive',
+    "Small Capitals": '"Arial Unicode Ms", Arial, Helvetica, Verdana, "Marcellus SC", sans-serif'
+  },
+
+  fontVariantMap: {
+    "Monospaced Serif": 'normal',
+    "Proportional Serif": 'normal',
+    "Monospaced Sans-Serif": 'normal',
+    "Proportional Sans-Serif": 'normal',
+    "Casual": 'normal',
+    "Cursive": 'normal',
+    "Small Capitals": 'small-caps'
+  },
+
+  fontSizeMap: {
+    "Small": {
+      "xs": (baseFontSize * 0.8) + "em",
+      "sm": (baseFontSize * 1.0) + "em",
+      "md": (baseFontSize * 1.2) + "em",
+      "lg": (baseFontSize * 1.4) + "em"
+    },
+    "Medium": {
+      "xs": (baseFontSize * 1.2) + "em",
+      "sm": (baseFontSize * 1.4) + "em",
+      "md": (baseFontSize * 1.6) + "em",
+      "lg": (baseFontSize * 1.8) + "em"
+    },
+    "Large": {
+      "xs": (baseFontSize * 1.6) + "em",
+      "sm": (baseFontSize * 1.8) + "em",
+      "md": (baseFontSize * 2.0) + "em",
+      "lg": (baseFontSize * 2.2) + "em"
+    },
+    "Extra Large": {
+      "xs": (baseFontSize * 2.0) + "em",
+      "sm": (baseFontSize * 2.2) + "em",
+      "md": (baseFontSize * 2.4) + "em",
+      "lg": (baseFontSize * 2.6) + "em"
+    }
+  },
+
+  textEnhancementMap: {
+    "Uniform": "none",
+    "Depressed": "1px 1px white",
+    "Raised": "-1px -1px white, -3px 0px 5px black",
+    "Shadow": "2px 2px 2px #1a1a1a"
+  },
+
+  setWindowBackgroundStyle: function(color, opacity) {
+    if (color == "Transparent") opacity = 0;
+    return {
+      backgroundColor: "rgba(" + this.colorMap[color] + "," + opacity + ")"
+    }
+  },
+
+  setTextStyle: function(color, opacity, fontType, fontSize, textEnhancement) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has 5 parameters which is greater than 4 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)